server/controllers/gym/editGym.ts

```diff
@@ -0,0 +1,20 @@
+import { NextFunction, Response, Request } from 'express';
+// import Sequelize from 'sequelize';
+import Gym from '../../database/models/gyms';
+import { CustomError, gymEditSchema } from '../../utils';
+
+export default async function editGymData(req: Request, res: Response, next: NextFunction) {
+  try {
+    await gymEditSchema.validateAsync(req.body);
+    const { id } = req.params;
+    const gym = await Gym.findByPk(id);
+    if (!gym) {
+      throw new CustomError('Gym not found', 404);
+    }
+    await gym.update(req.body);
```

Comment:
here id  I have a gym token, I'm able to update any gym data,
So we need first in the route itself to use `checkAuth('gym')`, that will put gym id in the request object,
then we have to check id in the request from checkAuth with id in the request body, If they are equivalent then user is authorized to update the data